src/lib.rs

```diff
@@ -31,25 +36,41 @@ mod inner {
         const BLOCK_SIZE: u8 = 8;
 
         #[new]
-        #[pyo3(signature = (string = Vec::new()))]
-        fn new(string: Vec<u8>) -> Self {
-            SeaHash { inner: string }
+        #[pyo3(signature = (data = Vec::new()))]
+        fn new(data: Vec<u8>) -> Self {
+            let mut inner = seahash::SeaHasher::new();
+            inner.write(data.as_slice());
+            SeaHash { inner }
         }
 
+        #[cfg(not(Py_3_11))]
         pub fn update(&mut self, obj: &[u8]) {
-            self.inner.extend_from_slice(obj)
+            self.inner.write(obj)
+        }
+
+        #[cfg(Py_3_11)]
+        pub fn update(&mut self, py: Python, obj: &PyAny) -> PyResult<()> {
+            match obj.extract() {
+                Ok(buf) => {
+                    self.inner.write(buf);
+                }
```

Comment:
Suggestion: This is likely better written using `if let Ok(buf) = obj.extract()`.